### PR TITLE
Add support for quadlets and networks

### PR DESCRIPTION
**Pull request by tobiasmcnulty**
Adds modern Podman deployment options while maintaining backwards compatibility.

New Features:

- Quadlet support (disabled by default; opt in via `podman_use_quadlets: yes`) - Uses Podman 4.4+ native systemd integration instead of `podman generate systemd`
- Network support (`podman_networks`) - Allows independent container management vs pod-based deployments

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -22,141 +72,216 @@
...
 
-- name: Build Docker-formatted image(s) with Podman
+- name: Build image(s) with Podman
   containers.podman.podman_image:
     name: "{{ item.0.image_name }}"
     path: "{{ item.0.dest_dir }}"
     build:
       format: docker
-      # Allow application to be aware of the deployed code version
       extra_args: "--build-arg PODMAN_COMMIT_SHA={{ item.1.after }}"
```
Comment (tobiasmcnulty) [inline tasks/main.yml:99]:
```suggestion
      # Allow application to be aware of the deployed code version
      extra_args: "--build-arg PODMAN_COMMIT_SHA={{ item.1.after }}"
```

File: tasks/main.yml [hunk 2]
```diff
@@ -13,6 +12,57 @@
...
+      {{ key }}={{ value }}
+      {% endfor %}
+    dest: "{{ podman_env_file_dir }}/{{ item.name }}.env"
+    mode: "0600"
+  loop: "{{ podman_containers | selectattr('env', 'defined') | rejectattr('state', 'defined') | list + podman_containers | selectattr('env', 'defined') | selectattr('state', 'defined') | rejectattr('state', 'eq', 'absent') | list }}"
+  loop_control:
+    label: "{{ item.name }}"
+  when: podman_use_env_files
+  register: env_files
+
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply tasks/main.yml:73]: @copilot open a PR to implement this feedback

File: tasks/main.yml [hunk 3]
```diff
@@ -13,6 +12,57 @@
...
+  when: podman_use_env_files
+
+- name: Write container environment files
+  ansible.builtin.copy:
+    content: |
+      {% for key, value in item.env.items() %}
+      {{ key }}={{ value }}
+      {% endfor %}
+    dest: "{{ podman_env_file_dir }}/{{ item.name }}.env"
+    mode: "0600"
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply tasks/main.yml]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/caktus/ansible-role-podman-containers/pull/2#discussion_r2805959214)

File: tasks/main.yml [hunk 4]
```diff
@@ -34,129 +81,272 @@
...
 
 - name: Create pod(s)
   containers.podman.podman_pod:
     name: "{{ item.name }}"
-    state: "{{ item.state | default('created') }}"
+    state: "{{ item.state | default(podman_use_quadlets | ternary('quadlet', omit)) }}"
     recreate: "{{ item.recreate | default(podman_recreate) | default(true) }}"
     publish: "{{ item.publish | default(omit) }}"
-    network: "{{ item.network | default(omit) }}"
+    network: "{{ item.network | default((podman_default_network is defined) | ternary([podman_default_network], omit)) }}"
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply tasks/main.yml]: @copilot open a PR to update or add a molecule test to confirm this behavior and implement a fix if needed ([feedback](https://github.com/caktus/ansible-role-podman-containers/pull/2#discussion_r2807547733)).

File: tasks/main.yml [hunk 5]
```diff
@@ -13,6 +12,54 @@
     mode: "0755"
   loop: "{{ podman_directories }}"
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply tasks/main.yml:34]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/caktus/ansible-role-podman-containers/pull/2#discussion_r2807547738)

File: README.md [hunk 6]
```diff
@@ -4,4 +4,73 @@ An Ansible role to run Podman containers on Ubuntu or RHEL-compatible servers.
...
+```
+
+### Test Scenarios
+
+| Scenario         | Mode    | What it tests                                                                       |
+| ---------------- | ------- | ----------------------------------------------------------------------------------- |
+| `default`        | Legacy  | Network creation, directory setup, systemd dirs                                     |
+| `legacy-pod`     | Legacy  | Pod with multiple containers, `podman generate systemd`                             |
+| `quadlets`       | Quadlet | Standalone containers on a shared network, `.container`/`.network` files, env files |
+| `quadlets-mixed` | Quadlet | Pod + standalone containers coexisting, `.pod`/`.container`/`.network` files        |
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply README.md]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/caktus/ansible-role-podman-containers/pull/2#discussion_r2807547741)

File: tasks/main.yml [hunk 7]
```diff
@@ -34,129 +83,283 @@
...
-  loop: "{{ podman_pods | map(attribute='name') }}"
-  when: podman_pod_inherit_hostname
+    - name: Create systemd unit file(s) for container(s)
+      command:
+        cmd: "podman generate systemd --new --files --name {{ item.name }}"
+        chdir: "{{ podman_user_home }}/.config/systemd/user/"
+      loop: "{{ podman_containers }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: podman_pod_name is not defined and not item.pod | default(false) and container.changed
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply tasks/main.yml]: @copilot provide a full investigation and report with suggested changes for this deprecated parameter, maximizing backwards compatibility where possible

**Comment by tobiasmcnulty** [comment]
@copilot open a PR to address the feedback in the review, adding or updating tests to catch the edge cases as needed: https://github.com/caktus/ansible-role-podman-containers/pull/2#pullrequestreview-3802061863

**Comment by tobiasmcnulty** [comment]
@copilot is it technically possible for a standalone container to inherit the host's hostname too, like a pod? This needs to be supported in legacy and quadlet mode. Add molecule scenarios to assess the current behavior and fix any issues discovered.

**Comment by tobiasmcnulty** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/caktus/ansible-role-podman-containers/pull/2#pullrequestreview-3803064247)

**Comment by tobiasmcnulty** [comment]
@copilot fix precommit

**Comment by tobiasmcnulty** [comment]
@copilot Open a PR to fix this issue identified in GHA:

```
 PLAY [Prepare instance] ********************************************************
  TASK [Wait for container to be ready] ******************************************
  Warning: : Reset is not implemented for this connection
  ok: [ubuntu]
  ok: [rhel]
  TASK [Ensure Ansible tmp directory exists] *************************************
  ok: [rhel]
  ok: [ubuntu]
  TASK [Wait for systemd to initialize] ******************************************
  ok: [rhel]
  ok: [ubuntu]
  TASK [Start network-online.target] *********************************************
  ok: [rhel]
  ok: [ubuntu]
  TASK [Start dbus service (RHEL/CentOS uses dbus-broker)] ***********************
  ok: [ubuntu]
  changed: [rhel]
  TASK [Start dbus service (Ubuntu/Debian uses dbus.service)] ********************
  skipping: [rhel]
  skipping: [ubuntu]
  TASK [Start user systemd instance for root (user@0)] ***************************
  fatal: [rhel]: FAILED! => {"attempts": 3, "changed": false, "msg": "Unable to start service user@0.service: Job for user@0.service failed because the control process exited with error code.\nSee \"systemctl status user@0.service\" and \"journalctl -xeu user@0.service\" for details.\n"}
  FAILED - RETRYING: [rhel]: Start user systemd instance for root (user@0) (3 retries left).
  changed: [ubuntu]
  FAILED - RETRYING: [rhel]: Start user systemd instance for root (user@0) (2 retries left).
  FAILED - RETRYING: [rhel]: Start user systemd instance for root (user@0) (1 retries left).
  PLAY RECAP *********************************************************************
  rhel                       : ok=5    changed=1    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0
  ubuntu                     : ok=6    changed=1    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
  CRITICAL Ansible return code was 2, command was: ansible-playbook --inventory /home/runner/.ansible/tmp/molecule.uSsk.default/inventory --skip-tags molecule-notest,notest /home/runner/work/ansible-role-podman-containers/ansible-role-podman-containers/molecule/prepare.yml
  ERROR    [default > prepare] Executed: Failed
  ERROR    Ansible return code was 2, command was: ansible-playbook --inventory /home/runner/.ansible/tmp/molecule.uSsk.default/inventory --skip-tags molecule-notest,notest /home/runner/work/ansible-role-podman-containers/ansible-role-podman-containers/molecule/prepare.yml
  ```

File: tasks/main.yml [hunk 8]
```diff
@@ -34,129 +102,317 @@
...
+  vars:
+    pod_containers_changed: >-
+      {{
+        container.results
+        | selectattr('item.pod', 'defined')
+        | selectattr('item.pod', 'equalto', item.item.name)
+        | selectattr('changed', 'equalto', true)
+        | list
+        | length > 0
+      }}
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply tasks/main.yml:398]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/caktus/ansible-role-podman-containers/pull/2#discussion_r2810317685)

**Review by copelco — APPROVED** [review]
